docs/extensions.adoc

```diff
@@ -228,6 +228,9 @@ class FlakyInegrationSpec extends Specification {
   @Retry(exceptions=[IOException])
   def onlyRetryIOException() { ... }
 
+  @Retry(condition = { failure.message.contains('foo') })
```

Comment:
Add also an instance example, and expand the explanation above regarding the condition.